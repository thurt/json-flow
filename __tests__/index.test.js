
const jsonFlow = require('../')

const sampleJson = {
  'test-list': [{
    id: 1,
    properties: { name: 'a' }
  }, {
    id: 2,
    properties: { name: 'b' }
  }],
  'empty-list': [],
  'voided-property': null
}

it('matches snapshot for undefined input', () => {
  expect(jsonFlow(undefined)).toMatchSnapshot()
})

it('matches snapshot for null input', () => {
  expect(jsonFlow(null)).toMatchSnapshot()
})

it('matches snapshot for sampleJson using 2 spaces', () => {
  expect(jsonFlow(sampleJson, 2)).toMatchSnapshot()
})

it('matches snapshot for sampleJson using 4 spaces', () => {
  expect(jsonFlow(sampleJson, 4)).toMatchSnapshot()
})

it('matches snapshot for sampleJson using tabs', () => {
  expect(jsonFlow(sampleJson, undefined, true)).toMatchSnapshot()
})


