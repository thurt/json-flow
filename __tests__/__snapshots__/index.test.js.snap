// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for null input 1`] = `"void"`;

exports[`matches snapshot for sampleJson using 2 spaces 1`] = `
"{
  'test-list': {
    'id': number,
    'properties': {
      'name': string,
    },
  }[],
  'empty-list': [],
  'voided-property': void,
}"
`;

exports[`matches snapshot for sampleJson using 4 spaces 1`] = `
"{
    'test-list': {
        'id': number,
        'properties': {
            'name': string,
        },
    }[],
    'empty-list': [],
    'voided-property': void,
}"
`;

exports[`matches snapshot for sampleJson using tabs 1`] = `
"{
	'test-list': {
		'id': number,
		'properties': {
			'name': string,
		},
	}[],
	'empty-list': [],
	'voided-property': void,
}"
`;

exports[`matches snapshot for undefined input 1`] = `"void"`;
